devicetypes/widomsrl/widom-smart-dry-contact.src/widom-smart-dry-contact.groovy

```diff
@@ -0,0 +1,548 @@
+                eventType: "ALERT",
+                name: "secureInclusion",
+                value: "failed",
+                displayed: true,
+        )
+    } else if (description == "updated") {
+        return null
+    } else {
+        deviceId = description.split(", ")[0]
+        deviceId = deviceId.split(":")[1]
+        logging("deviceId- ${deviceId}")
+        deviceIdEvent(deviceId, true, true)
+        def cmd = zwave.parse(description, cmdVersions())
+        if (cmd) {
+            logging("Parsed: ${cmd}")
+            zwaveEvent(cmd)
+        }
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+    def encapsulatedCommand = cmd.encapsulatedCommand(cmdVersions())
+    if (encapsulatedCommand) {
+        logging("Parsed SecurityMessageEncapsulation into: ${encapsulatedCommand}")
+        zwaveEvent(encapsulatedCommand)
+    } else {
+        logging("Unable to extract Secure command from $cmd","warn")
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+    def version = cmdVersions()[cmd.commandClass as Integer]
+    def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+    def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+    if (encapsulatedCommand) {
+        logging("Parsed Crc16Encap into: ${encapsulatedCommand}")
+        zwaveEvent(encapsulatedCommand)
+    } else {
+        logging("Unable to extract CRC16 command from $cmd","warn")
+    }
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+    def encapsulatedCommand = cmd.encapsulatedCommand(cmdVersions())
+    if (encapsulatedCommand) {
+        logging("Parsed MultiChannelCmdEncap ${encapsulatedCommand}")
+        zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+    } else {
+        logging("Unable to extract MultiChannel command from $cmd","warn")
+    }
+}
+
+private logging(text, type = "debug") {
+    if (settings.logging == "true" || type == "warn") {
+        log."$type" "${device.displayName} - $text"
+    }
+}
+
+private secEncap(physicalgraph.zwave.Command cmd) {
+    logging("encapsulating command using Secure Encapsulation, command: $cmd","info")
+    zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private crcEncap(physicalgraph.zwave.Command cmd) {
+    logging("encapsulating command using CRC16 Encapsulation, command: $cmd","info")
+    zwave.crc16EncapV1.crc16Encap().encapsulate(cmd).format()
+}
+
+private multiEncap(physicalgraph.zwave.Command cmd, Integer ep) {
+    logging("encapsulating command using MultiChannel Encapsulation, ep: $ep command: $cmd","info")
+    zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint:ep).encapsulate(cmd)
+}
+
+private encap(physicalgraph.zwave.Command cmd, Integer ep) {
+    encap(multiEncap(cmd, ep))
+}
+
+private encap(List encapList) {
```

Comment:
Please, could You point where do You use this method with a list?

Reply:
Method removed